docs/DafnyRef/UserGuide.md

```diff
@@ -51,129 +96,315 @@ verification failure in a file that is being checked.
-
+For example, legacy `dafny` accepts `/noVerify` and `-noVerify`, while the current command-oriented option is `--no-verify`.
+ 
 If no command is given, then the command-line is presumed to use old-style syntax, so any previously 
 written command-line will still be valid.
 
-The following commands are recognized. In each case all files listed are parsed and typechecked.
-- `dafny verify` -- verifies the listed files. Although the Dafny program being considered
+`dafny` recognizes the commands described in the following subsections. The most commonly used
+are [`dafny verify`](#sec-dafny-verify), [`dafny build`](#sec-dafny-build), and [`dafny run`](#sec-dafny-run).
+
+The command-line also expects the following:
+- Files are designated by absolute paths or paths relative to the current
+working directory. A command-line argument not matching a known option is considered a filepath, and likely one
+with an unsupported suffix, provoking an error message..
+- Files containing dafny code must have a `.dfy` suffix.
+- There must be at least one `.dfy` file.
+- The command-line may contain other kinds of files appropriate to
+the language that the Dafny files are being compiled to. The kind of file is determined by its suffix.
+- Escape characters are determined by the shell executing the command-line.
+- The option `--` means that all subsequent command-line arguments are not options to the dafny tool; they are either files or arguments to the `dafny run` command.
+- If an option is repeated (e.g., with a different argument), then the later instance on the command-line supersedes the earlier instance, with just a few options accumulating arguments.
+- If an option takes an argument, the option name is followed by a `:` or `=` or whitespace and then by the argument value; if the argument itself contains white space, the argument must be enclosed in quotes.
+- Boolean options can take the values `true` and `false` (or any case-insensitive version of those words). For example, the value of `--no-verify` is by default `false` (that is, do verification). It can be set to true (no verification) using `--no-verify`, `--no-verify:true`, `--no-verify=true`, `--noverify true`; it can be set false (do verification) using `--no-verify:false` or `--no-verify=false` or `--no-verify false`.
+- There is a potential ambiguity when the form `--option value` is used if the value is optional (such as for boolean values). In such a case an argument afer an option (that does not have an argument given with `:` or `=`) is interpreted as the value if it is indeed a valid value for that option. However, better style advises always using a ':' or '=' to set option values.
+No valid option values in dafny look like filenames or begin with `--`.
+
+
+Options may change over time; `dafny <command> --help` lists those options for a given \<command\> that are implemented in your version of dafny.
```

Comment:
> Options may change over time

Why mention this? We try not to change them, unless they are marked with (experimental)

Reply:
Took out that phrase, though the set of options certainly does evolve over time.